sensing/autoware_pointcloud_preprocessor/src/concatenate_data/concatenate_and_time_sync_node.cpp

```diff
@@ -242,36 +241,36 @@ void PointCloudConcatenateDataSynchronizerComponent::cloud_callback(
       collector_matching_strategy_->match_cloud_to_collector(cloud_collectors_, matching_params);
   }
 
-  bool process_success = false;
-  if (cloud_collector.has_value()) {
-    auto collector = cloud_collector.value();
-    if (collector) {
-      cloud_collectors_lock.unlock();
-      process_success = collector->process_pointcloud(topic_name, input_ptr);
-    }
+  if (cloud_collector.has_value() && cloud_collector.value()) {
+    selected_collector = cloud_collector.value();
   }
 
   // Didn't find matched collector
-  if (!process_success) {
-    // Reuse the collector if the status is IDLE
-    std::shared_ptr<CloudCollector> selected_collector = nullptr;
-
+  if (!selected_collector || selected_collector->get_status() == CollectorStatus::Finished) {
+    // Reuse a collector if one is in the IDLE state
     auto it = std::find_if(
       cloud_collectors_.begin(), cloud_collectors_.end(),
       [](const auto & collector) { return collector->get_status() == CollectorStatus::Idle; });
 
     if (it != cloud_collectors_.end()) {
       selected_collector = *it;
+    } else {
+      auto oldest_it = find_and_reset_oldest_collector();
+      if (oldest_it != cloud_collectors_.end()) {
+        selected_collector = *oldest_it;
+      } else {
+        // Handle case where no suitable collector is found
+        RCLCPP_WARN(get_logger(), "No available CloudCollector in IDLE state.");
```

Comment:
This should probably be an error (the invariant of the program, that there is at least 1 idle one after `find_and_reset_oldest_collector` is broken)
```suggestion
        RCLCPP_ERROR(get_logger(), "No available CloudCollector in IDLE state.");
```

Reply:
addressed in https://github.com/autowarefoundation/autoware.universe/pull/10082/commits/0c5de07353ecceb159c26598f9531dd8939a2351